### DISCUSSION TITLE
[BUG] - "published_at" queries not working in storefront for products query

### DISCUSSION THREAD

**Discussion by joelpierre** [Shopify/storefront-api-feedback#233] [Help] [answered]
When I use the following query:

```gql
{
  products(first: 250, query: "published_at:>='2024-02-12'") {
    nodes {
      id
      availableForSale
      title
      tags
      publishedAt
      variants(first: 1) {
        nodes {
          id
        }
      }
    }
    pageInfo {
      endCursor
    }
  }
}
```

<img width="1275" alt="Screenshot 2024-02-13 at 13 21 09" src="https://github.com/Shopify/storefront-api-feedback/assets/8969851/6f7f33bf-40a5-4576-b93b-b7a9a0132a62">

The response contains ALL products in my catalogue when it should only be the last ~days worth of products if I change `published_at` to `created_at` then the query works as expected... So it seems there is an issue fetching `published_at` products.

<img width="1271" alt="Screenshot 2024-02-13 at 13 21 55" src="https://github.com/Shopify/storefront-api-feedback/assets/8969851/3547e1df-baa8-4e3f-8489-ea2337bb000a">

Problem here though is I want to fetch recently published products because published products do not necessarily mean they were recently created.

**Comment by corygibbons** [comment, marked as answer]
I don't believe `published_at` is supported in the query argument.
Per docs as of `2024-01` it only supports `created_at` and `updated_at` as far as dates are concerned.

Full list here: https://shopify.dev/docs/api/storefront/2024-01/queries/products#argument-query

**Reply by joelpierre** [reply]
Thanks @corygibbons yeah, I realised that but didn't come back to this discussion 🙈 however still noticing anomalies with `created_at`. But slightly more bearable. I wish `published_at` was supported though...
